src/App.tsx

```diff
@@ -1,223 +1,294 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import {
+  getTodos,
+  createTodos,
+  deleteTodos,
+  UpdateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { TodoLIst } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Error } from './components/Error';
+import { Header } from './components/Header';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [errorAdd, setErrorAdd] = useState(false);
+  const [errorRemove, setErrorRemove] = useState(false);
+  const [errorUpdate, setErrorUpdate] = useState(false);
+  const [errorEmptyTitile, setErrorEmptyTitile] = useState(false);
+  const [typeFilter, setTypeFilter] = useState('All');
+  const [hidden, setHidden] = useState(true);
+  const [isAdding, setIsAdding] = useState<Todo | null>(null);
+  const [clearLoader, setClearLoader] = useState(false);
+  const [loadingAllTodos, setLoadingAllTodos] = useState(false);
+  const [isLoading, setIsLoading] = useState<Todo | null>(null);
+
+  const handleHidden = useCallback(
+    (value: boolean) => {
+      setHidden(value);
+    }, [],
+  );
+
+  const handleLoading = useCallback((value: Todo | null) => {
+    setIsLoading(value);
+  }, []);
+
+  const handleTypeFilter = useCallback((type: string) => {
+    setTypeFilter(type);
+  }, []);
+
+  const hanldeUpdateError = useCallback((value: boolean) => {
+    setErrorUpdate(value);
+  }, []);
+
+  const handlerRemoveError = useCallback((value: boolean) => {
+    setErrorRemove(value);
+  }, []);
+
+  const handlerNewTitle = useCallback((value: string) => {
+    setNewTodoTitle(value);
+  }, []);
+
+  const foundTodoList = useCallback((u:User) => {
```

Comment:
probably you don't want to found smth here, just want to get updated data from server?)
let's think about naming for this function